scripts/sct_apply_transfo.py

```diff
@@ -267,40 +271,34 @@ def apply(self):
 
 # MAIN
 # ==========================================================================================
-def main(args=None):
-
-    # check user arguments
-    if not args:
-        args = sys.argv[1:]
+def main(arguments):
```

Comment:
Usually a `main()` function receives command-line arguments but not digested argparse arguments. So I would recommend to do the parsing inside main. You can then call another function that receives the args.

Reply:
Ok sounds good. Does something along [these lines](https://github.com/neuropoly/spinalcordtoolbox/blob/d19a0813a896230d601a4f9eaec5dca07e43ad5b/scripts/sct_analyze_lesion.py#L507-L509) follow what you had in mind?